### PR TITLE
Added type definitions components.

### LINKED ISSUE #370: Type definitions for all components for intellisense to work.

### PR DESCRIPTION
**Pull request by kamolesh97**
closes #370

### CODE REVIEW

**Comment by kamolesh97** [comment]
@sangameshsomawar _a please review and merge.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+    RNTextInputIOSProps,
+    StyleProps,
+    TypographyProps {}
+interface FlatListProps extends RNFlatListProps, StyleProps {}
+interface ScrollViewProps extends RNScrollViewProps, StyleProps {}
+interface TouchableProps extends RippleProps, StyleProps, ButtonStyleProps {
+  children?: React.ReactNode;
+}
+
+interface AccordianProps extends ViewProps {
```
Comment (sangameshsomawar) [inline index.d.ts:92]: Please fix spell error for all Accordions
```suggestion
interface AccordionProps extends ViewProps {
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+  header?: () => void;
```
Comment (sangameshsomawar) [inline index.d.ts:93]: the header is component, right?
```suggestion
  header?: React.ReactNode;
```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+
+interface CardProps extends TouchableOpacityProps, StyleProps {
+  elevation?: number;
+}
+
+type CarouselProps = {
+  itemArray: Array<any>;
+  renderItem: () => void;
+  carouselRef?: React.RefObject<Carousel>;
+  onSnapToItem?: () => void;
```
Comment (sangameshsomawar) [inline index.d.ts:204]: @kamolesh97 search all occurrences of `() => void` and check it returns something or we need to use React.Node if it accepts component.

This returns the number.
```suggestion
  onSnapToItem?: () => number;
```

**Review by sangameshsomawar — CHANGES_REQUESTED** [review]
@kamolesh97 _a Looks awesome. Please refer to the loom video for minor suggestions.

https://user-images.githubusercontent.com/93119254/186116320-693b947e-1930-4553-997d-e4a8673765c4.mp4

**Comment by kamolesh97** [comment]
> @kamolesh97 _a Looks awesome. Please refer to the loom video for minor suggestions.
> 
>  Screenshot.-.23.August.2022.mp4

@sangameshsomawar I too thought of adding Javadoc comments for props, but was not sure if I should add it now, maybe I can create a separate PR for that?

File: index.d.ts (hunk 2 above)
Comment (kamolesh97) [reply index.d.ts:93]: @sangameshsomawar header actually accepts a function.
https://github.com/bigbinary/neeto-ui-rn/blob/1856bc512ab06aaaceab03ab2b640db2005eee02/src/components/Accordion.js#L152

**Comment by kamolesh97** [comment]
@sangameshsomawar _a please check the latest commit, for Javadoc comments I work on a separate PR.

**Pull request merged**